src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/markup_rope_inputs.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "markup_rope_inputs.hpp"
+
+#include <unordered_set>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/constant_folding.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/rotary_positional_embeddings.hpp"
+#include "transformations/rt_info/disable_fp16_compression.hpp"
+#include "transformations/utils/gen_pattern.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::intel_cpu::MarkUpRopeInputs::MarkUpRopeInputs() {
+    MATCHER_SCOPE(MarkUpRopeInputs);
+    using namespace ov::pass::pattern;
+    using namespace ov::gen_pattern;
+    auto cos_tab = any_input();
+    auto sin_tab = any_input();
+    auto rope = makePattern<ov::op::internal::RoPE>({any_input(), cos_tab, sin_tab});
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto cos_input_node = pattern_map.at(cos_tab).get_node_shared_ptr();
+        auto sin_input_node = pattern_map.at(sin_tab).get_node_shared_ptr();
+        auto bfs_markup = [&](std::shared_ptr<ov::Node>& input) {
+            nodes.push_back(input);
+            while (!nodes.empty()) {
+                auto curr_node = nodes.front();
+                nodes.pop_front();
+                visited.insert(curr_node);
+                // visit cur node
+                ov::disable_fp16_compression(curr_node);
```

Comment:
just for the record, the name `disable_fp16_compression` is misleading, the actual behavior of this function is to 

> marks operation as prohibited to convert to lower precision (e.g. to FP16) and they should be inferred precisely in the original precision.  from `class TRANSFORMATIONS_API DisableFP16Compression`

so bf16 is also prohibited.